v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/transforms/RangeBoundaryTransform.java

```diff
@@ -59,14 +60,18 @@ public abstract class RangeBoundaryTransform
 
   @Override
   public PCollection<Range> expand(PCollection<ColumnForBoundaryQuery> input) {
-    return input.apply(
+    SingleOutput<ColumnForBoundaryQuery, Range> parDo =
         ParDo.of(
             new RangeBoundaryDoFn(
                 dataSourceProviderFn(),
                 dbAdapter(),
                 tableName(),
                 partitionColumns(),
-                boundaryTypeMapper())));
+                boundaryTypeMapper()));
+    if (boundaryTypeMapper() != null) {
+      parDo = parDo.withSideInputs(boundaryTypeMapper().getCollationMapperView());
```

Comment:
Majority of the code is per-collation.
The place where it's inovked in readWithUniformParition is per-table.
I have a todo to make this per DB or perShard.

The query as such takes ~ 30 Seconds and currently too it will run in parallel so from overhead POV its not a big problem.